UriGenerator.php

```diff
@@ -106,4 +123,16 @@ public function resolveConflict(UriContext $uriContext)
 
         return $uri;
     }
+
+    /**
+     * Configure options which apply to each token provider
+     *
+     * @param OptionsResolver
+     */
+    private function configureGlobalOptions(OptionsResolver $optionsResolver)
+    {
+        $optionsResolver->setDefaults(array(
+            'collapse_empty' => false,
```

Comment:
Well, imo:

* If a token is empty, it's trailing slash is stripped
* By default, a token cannot be empty. An empty token throws an exception
* You can change this behaviour by overriding `allow_empty`, which will allow an empty token

I think they are 2 generic and seperate processes. One is to always remove trailing slashes when a token is empty. The other is to allow or not to allow empty tokens. Only the second process can be changed by this option